test_libs/pyspec/eth2spec/test/epoch_processing/test_process_registry_updates.py

```diff
@@ -5,10 +5,34 @@
     is_active_validator,
     process_registry_updates
 )
+from eth2spec.phase0.state_transition import state_transition
+from eth2spec.test.helpers.block import build_empty_block_for_next_slot, sign_block
 from eth2spec.test.helpers.state import next_epoch
 from eth2spec.test.context import spec_state_test
 
 
+def run_process_registry_updates(state, valid=True):
+    """
+    Run ``process_crosslinks``, yielding:
+      - pre-state ('pre')
+      - post-state ('post').
+    If ``valid == False``, run expecting ``AssertionError``
+    """
+    # transition state to slot before state transition
+    slot = state.slot + (spec.SLOTS_PER_EPOCH - state.slot % spec.SLOTS_PER_EPOCH) - 1
+    block = build_empty_block_for_next_slot(state, signed=False)
+    block.slot = slot
+    sign_block(state, block)
+    state_transition(state, block)
+
+    # cache state before epoch transition
+    spec.cache_state(state)
+
```

Comment:
My gut is to run the following functions prior to yield `pre` to get the state to the exact point in the normal state transition. What do you think?

```python
    process_justification_and_finalization(state)
    process_crosslinks(state)
    process_rewards_and_penalties(state)
```

Reply:
Have we formalized dependencies in any way? Some clients process them in different order / parallel.

Reply:
We have not, but epoch processing cannot in general be processed out of order. Rewards/penalties can be processed in parallel and most of the block operations can be validated in parallel but not fully processed.

`process_registry_updates` certainly has a dependency on rewards/penalties (which has a dependency on justification/finality/crosslinks) due to the ejection condition `validator.effective_balance <= EJECTION_BALANCE`.

In isolation, it is not strictly incorrect to not run the prior functions in this test, but it is creating a more contrived state rather than what would actually be seen in the wild